### PR TITLE
feat: new Obj-C annotations

### PR DESCRIPTION
**Pull request by danil-pavlov**

### CODE REVIEW

File: docs/topics/native/native-objc-interop.md [hunk 1]
```diff
@@ -16,6 +16,25 @@ Kotlin modules can be used in Swift/Objective-C code if compiled into a
 framework ([see here for how to declare binaries](multiplatform-build-native-binaries.md#declare-binaries)).
 See [Kotlin Multiplatform Mobile Sample](https://github.com/Kotlin/kmm-basic-sample) for an example.
 
+### Hiding Kotlin declarations
+
+You can choose not to export Kotlin declaration to Objective-C and Swift with special annotations:
+
+* `@HiddenFromObjC` hides a Kotlin declaration from Objective-C. The annotation instructs the Kotlin compiler not to export
```
Comment (SvyatoslavScherbina) [inline docs/topics/native/native-objc-interop.md:23]: > hides a Kotlin declaration from Objective-C

And, consequently, Swift

File: docs/topics/native/native-objc-interop.md [hunk 2]
```diff
@@ -16,6 +16,25 @@ Kotlin modules can be used in Swift/Objective-C code if compiled into a
 framework ([see here for how to declare binaries](multiplatform-build-native-binaries.md#declare-binaries)).
 See [Kotlin Multiplatform Mobile Sample](https://github.com/Kotlin/kmm-basic-sample) for an example.
 
+### Hiding Kotlin declarations
+
+You can choose not to export Kotlin declaration to Objective-C and Swift with special annotations:
+
+* `@HiddenFromObjC` hides a Kotlin declaration from Objective-C. The annotation instructs the Kotlin compiler not to export 
+   a function or property to Objective-C, making your Kotlin code more Objective-C-friendly.
```
Comment (SvyatoslavScherbina) [inline docs/topics/native/native-objc-interop.md:24]: Objective-C/Swift-friendly

File: docs/topics/native/native-objc-interop.md [hunk 3]
```diff
@@ -16,6 +16,25 @@ Kotlin modules can be used in Swift/Objective-C code if compiled into a
 framework ([see here for how to declare binaries](multiplatform-build-native-binaries.md#declare-binaries)).
 See [Kotlin Multiplatform Mobile Sample](https://github.com/Kotlin/kmm-basic-sample) for an example.
 
+### Hiding Kotlin declarations
+
+You can choose not to export Kotlin declaration to Objective-C and Swift with special annotations:
```
Comment (koshachy) [inline docs/topics/native/native-objc-interop.md:21]: If you don't want to export Kotlin declarations to Objc and Swift, use special annotations:

WDYT?

File: docs/topics/native/native-objc-interop.md [hunk 4]
```diff
@@ -16,6 +16,25 @@ Kotlin modules can be used in Swift/Objective-C code if compiled into a
...
+
+* `@HiddenFromObjC` hides a Kotlin declaration from Objective-C and, consequently, Swift. The annotation instructs the
```
Comment (koshachy) [inline docs/topics/native/native-objc-interop.md:23]: Do we need "consequently" here explicitly?

File: docs/topics/native/native-objc-interop.md [hunk 5]
```diff
@@ -16,6 +16,25 @@ Kotlin modules can be used in Swift/Objective-C code if compiled into a
...
+  Kotlin compiler not to export a function or property to Objective-C, making your Kotlin code more Objective-C/Swift-friendly.
```
Comment (koshachy) [inline docs/topics/native/native-objc-interop.md:24]: Maybe we should rephrase it to "The annotation disables a function or prop export to objc" ? 

WDYT?

File: docs/topics/native/native-objc-interop.md [hunk 6]
```diff
@@ -16,6 +16,25 @@ Kotlin modules can be used in Swift/Objective-C code if compiled into a
...
+* @ShouldRefineInSwift` helps to replace a Kotlin declaration with a wrapper written in Swift. The annotation instructs
```
Comment (koshachy) [inline docs/topics/native/native-objc-interop.md:25]:
```suggestion
* `@ShouldRefineInSwift` helps to replace a Kotlin declaration with a wrapper written in Swift. The annotation instructs
```

Comment (koshachy) [reply docs/topics/native/native-objc-interop.md:25]: helps to replace -> replaces?

Comment (koshachy) [reply docs/topics/native/native-objc-interop.md:25]: replaces a Kotlin declaration with a Swift wrapper?

File: docs/topics/native/native-objc-interop.md [hunk 7]
```diff
@@ -16,6 +16,25 @@ Kotlin modules can be used in Swift/Objective-C code if compiled into a
...
+  the Kotlin compiler to mark a function or property as `swift_private` in the generated Objective-C API.
+  Such declarations get the `__` prefix, which makes them invisible from Swift.
+
+  You can still use these declarations in your Swift code to create a Swift-friendly API, but they won't be suggested in
+  the Xcode autocomplete, for example.
```
Comment (koshachy) [inline docs/topics/native/native-objc-interop.md:30]: do we need these "for example"?
Nothing changes if we just omit it.

File: docs/topics/native/native-objc-interop.md [hunk 8]
```diff
@@ -117,6 +155,10 @@ The methods of `kotlin.Any` (`equals()`, `hashCode()` and `toString()`) are mapp
 to the methods `isEquals:`, `hash` and `description` in Objective-C, and to the method
 `isEquals(_:)` and the properties `hash`, `description` in Swift.
 
+You can specify a more idiomatic name in Swift or Objective-C, instead of renaming the Kotlin declaration.
+Use the `@ObjCName` annotation that instructs the Kotlin compiler to use a custom Objective-C and Swift name for methods
+or parameters. Note that using this annotation requires [opt-in](opt-in-requirements.md).
```
Comment (koshachy) [inline docs/topics/native/native-objc-interop.md:160]: Shouldn't it be a note?

```
> Using this annotation requires opt-in
> 
{type="note"}
```

**Review by koshachy — CHANGES_REQUESTED** [review]

File: docs/topics/native/native-objc-interop.md (hunk 6 above)
Comment (SvyatoslavScherbina) [reply docs/topics/native/native-objc-interop.md:25]: No: https://github.com/JetBrains/kotlin-web-site/pull/3284#discussion_r1032564803
This annotation makes Kotlin declaration "less visible" to Swift (but still keeps it accessible). It helps in that particular way, but doesn't replace anything.

**Review by SvyatoslavScherbina — APPROVED** [review]

**Review by koshachy — APPROVED** [review]
